web-wallet/src/__mocks__/AddressSyncer.js

```diff
@@ -0,0 +1,107 @@
+import { AddressSyncer } from "$lib/vendor/w3sper.js/src/network/syncer/address";
+
+import { cacheUnspentNotes } from "$lib/mock-data";
+
+import notesArrayToMap from "$lib/wallet/notesArrayToMap";
+
+/**
+ * @template T
+ * @param {Array<T>} arr
+ * @param {number} n
+ */
+function* chunkGenerator(arr, n) {
```

Comment:
Looks like this could result in an unbalanced distribution if the division is not even (ie. the array length isn’t perfectly divisible by `n`).

I guess it's not a big deal for the purpose of the mock.

Reply:
You did good mentioning it but, yes, it is the intended behaviour: the function wasn't meant to create an even distribution, but rather a predictable amount of "chunks" (even if empty).